File: 392/main.js
isSubsequence("aaaaaa", "bbaaaa")

function isSubsequence(s, t) {
    let sPointer = 0;
    let tPointer = 0;

    while (sPointer < s.length && tPointer < t.length) {
        if (s[sPointer] === t[tPointer]) {
            sPointer++;
        }
        tPointer++;
    }

    return sPointer === s.length;
}
 

/**
 * 
        const subString = t.substring(lastIndex);

        let i = subString.indexOf(c); 

        

        if (i == -1) {
            return false;
        } 

        lastIndex = i;
         
        console.log(c, i , subString, lastIndex);
 */

File: 392/main.go
package main

import "fmt"

func main() {
	s1, t1 := "abc", "ahbgdc"
	s2, t2 := "axc", "ahbgdc"

	fmt.Println(isSubsequence(s1, t1)) // Output: true
	fmt.Println(isSubsequence(s2, t2)) // Output: false
}

func isSubsequence(s string, t string) bool {
	sPointer, tPointer := 0, 0

	for sPointer < len(s) && tPointer < len(t) {
		if s[sPointer] == t[tPointer] {
			sPointer++
		}
		tPointer++
	}

	return sPointer == len(s)
}
